File: slizaa-hierarchicalgraph/core/org.slizaa.hierarchicalgraph.core.testfwk/src/test/java/org/slizaa/hierarchicalgraph/core/testfwk/mapstruct/LookupNodeTest.java
package org.slizaa.hierarchicalgraph.core.testfwk.mapstruct;

import org.eclipse.emf.ecore.util.EcoreUtil;
import org.junit.ClassRule;
import org.junit.Test;
import org.slizaa.hierarchicalgraph.core.model.HGNode;
import org.slizaa.hierarchicalgraph.core.model.HierarchicalgraphPackage;
import org.slizaa.hierarchicalgraph.core.testfwk.XmiBasedGraph;
import org.slizaa.hierarchicalgraph.core.testfwk.XmiBasedTestGraphProviderRule;

import static org.assertj.core.api.Assertions.assertThat;

public class LookupNodeTest {

  @ClassRule
  public static XmiBasedTestGraphProviderRule _graphProvider = new XmiBasedTestGraphProviderRule(XmiBasedGraph.MAP_STRUCT);

    @Test
  public void testLookup() {
    EcoreUtil.getAllContents(_graphProvider.rootNode(), false).forEachRemaining((c) -> {
      if (HierarchicalgraphPackage.eINSTANCE.getHGNode().isInstance(c)) {
        HGNode node = (HGNode) c;
        assertThat(node.getIdentifier()).isNotNull();
        assertThat(_graphProvider.rootNode().lookupNode(node.getIdentifier())).isEqualTo(node);
      }
    });
  }
}

File: slizaa-hierarchicalgraph/core/io.hierograph.hierarchicalgraph.core.model/src/test/kotlin/io/hierograph/hierarchicalgraph/core/model/LookupNodeTest.kt
package io.hierograph.hierarchicalgraph.core.model

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class LookupNodeTest {

    private lateinit var g: SimpleTestGraph

    @BeforeEach
    fun setup() { g = SimpleTestGraph() }

    @Test
    fun `lookupNode returns correct node`() {
        val found = g.root.lookupNode(g.a1.identifier)
        assertThat(found).isSameAs(g.a1)
    }

    @Test
    fun `lookupNode returns correct leaf node`() {
        val found = g.root.lookupNode(g.b3.identifier)
        assertThat(found).isSameAs(g.b3)
    }

    @Test
    fun `lookupNode returns null for unknown identifier`() {
        val found = g.root.lookupNode(-1L)
        assertThat(found).isNull()
    }
}
